src/components/JoinList.js

```diff
@@ -1,13 +1,10 @@
-import React, { useEffect } from 'react';
+import React from 'react';
 import { Link } from 'react-router-dom';
 import '../pages/HomePage.css';
-import useListToken from '../useListToken';
+import useListToken, { getCurrentToken } from '../useListToken';
 
 const JoinList = () => {
-  const { token, saveToken } = useListToken();
-  useEffect(() => {
-    saveToken(token);
-  }, [saveToken, token]);
+  const { saveToken } = useListToken();
 
   const handleChange = event => {
```

Comment:
We shouldn't be calling `saveToken` in the `onChange`. This writes to `localStorage` on every key stroke and tries to fetch the list with that token—something we don't want.

```suggestion
   const [joinToken, setJoinToken] = useState('');

   const handleChange = event => {
     setJoinToken(event.target.value)
   }
   
   const handleSubmit = event => {
     event.preventDefault();
     saveToken(joinToken)
   }
```

Reply:
This needs to be addressed before we move on. Otherwise, we have the same problem we had before of hitting the database on every keystroke by the user.

Reply:
Nikema, let's just double check that we're not generating database reads with every keystroke, then I'm not concerned with this. Thanks!